server/controllers/userController.mjs

```diff
@@ -207,40 +207,50 @@ export const resetPassword = async (req, res) => {
 
 export const updateUserProfile = async (req, res) => {
   const { uid } = req.params; // User ID from URL params
-  const { name, profilePictureUrl } = req.body; // New name and profile picture URL
+  const { name, email, profilePictureUrl } = req.body; // New profile fields
 
-  // Check if UID is provided
   if (!uid) {
     return res.status(400).json({ error: 'User ID is required' });
   }
 
-  // Prepare the updates object
   const updates = {};
+
   if (name) {
     if (typeof name !== 'string' || name.trim() === '') {
       return res.status(400).json({ error: 'Invalid name' });
     }
-    updates.name = name; // Update name field in Firebase Authentication
+    updates.displayName = name;
+    updates.name = name; // For Firestore
```

Comment:
We don't have a displayName field in the userModel.mjs file. Here are the fields for the user model:
 " name,
    email,
    userType,
    profilePictureUrl = null, // Default image URL
    createdAt = new Date(),"

Reply:
addressed!